packages/edit-site/src/components/post-fields/index.js

```diff
@@ -164,30 +159,10 @@ function usePostFields( viewType ) {
 						? item.title
 						: item.title?.raw,
 				render: ( { item } ) => {
-					const addLink =
-						[ LAYOUT_TABLE, LAYOUT_GRID ].includes( viewType ) &&
-						item.status !== 'trash';
 					const renderedTitle =
 						typeof item.title === 'string'
 							? item.title
 							: item.title?.rendered;
-					const title = addLink ? (
-						<Link
-							params={ {
-								postId: item.id,
-								postType: item.type,
-								canvas: 'edit',
-							} }
-						>
-							{ decodeEntities( renderedTitle ) ||
-								__( '(no title)' ) }
-						</Link>
```

Comment:
Is there a way to still keep the actual link?

Reply:
https://github.com/WordPress/gutenberg/blob/5c7b0c45c8f68f863521f3312c161d8b3a8d9d83/packages/dataviews/src/dataviews-layouts/table/index.tsx#L124-L146

We could wrap the field render in a `Link` component, but I'm not sure. The primary field might contain another `a` element, which could trigger a `ValidateDomNesting` warning.

Reply:
Yeah, that's not ideal!
I am still leaning towards the fact that this should continue to be handled on a field level and maybe we only expose the `onItemClick` function that can be used as a callback with the `field` being clicked.

In order for us to do this, we did have to pass down a `onClick` function to the field render method.
Alluding to @oandregal [last thought](https://github.com/WordPress/gutenberg/pull/66365#issuecomment-2455329276), maybe we want to pass down a `onEvent` callback that takes the event type `click` or something else, and the `field`.
The render method within the field can decide how to handle this.